File: problems/python/MeetingScheduler/src/models/notification.py
from dataclasses import dataclass
from datetime import datetime

from enums.notification_type import NotificationType


@dataclass(frozen=True)
class Notification:
    notification_id: int
    notification_type: NotificationType
    content: str
    creation_date: datetime

    def send(self, user: "User", meeting: "Meeting") -> None:
        print(
            f"[{self.notification_id}] {self.notification_type.label} to {user.name}"
            f" <{user.email}> — {self.content} ({meeting.subject}) @ {self.creation_date}"
        )

File: problems/java/MeetingScheduler/src/models/Notification.java
package models;

import enums.NotificationType;

import java.time.LocalDateTime;

public class Notification {

    private final int id;
    private final NotificationType type;
    private final String content;
    private final LocalDateTime creationDate;

    public Notification(int id, NotificationType type, String content, LocalDateTime creationDate) {
        this.id = id;
        this.type = type;
        this.content = content;
        this.creationDate = creationDate;
    }

    public void send(User user, Meeting meeting) {
        System.out.println("[" + id + "] " + type.getLabel() + " to " + user.getName()
                + " <" + user.getEmail() + "> — " + content + " (" + meeting.getSubject()
                + ") @ " + creationDate);
    }
}
